cassandra/pool.py

```diff
@@ -360,18 +363,45 @@ def __init__(self, host, host_distance, session):
             return
 
         log.debug("Initializing connection for host %s", self.host)
-        self._connection = session.cluster.connection_factory(host.endpoint)
+        first_connection = session.cluster.connection_factory(host.endpoint)
+        log.debug("first connection created for shard_id=%i", first_connection.shard_id)
+        self._connections[first_connection.shard_id] = first_connection
         self._keyspace = session.keyspace
+
         if self._keyspace:
-            self._connection.set_keyspace_blocking(self._keyspace)
+            first_connection.set_keyspace_blocking(self._keyspace)
+
+        if first_connection.sharding_info:
+            self.host.sharding_info = weakref.proxy(first_connection.sharding_info)
+            for _ in range(first_connection.sharding_info.shards_count * 2):
```

Comment:
Why it's aggressive ? i.e. once we get to the number we need, we'll stop.
cause we might open twice as much connection, and not get to all of the shards ?

Reply:
No, the other way around. ;)
I think assuming that after you attempt to open <shard_num>x2 connections you will have all shards populated is probably too optimistic ("aggressive").

Reply:
Oh, this aggressive :)

Yeah maybe 3 would be better lucky number here.

since you one error and you not necessarily do the cycle (and no guarantees for ordering anyhow)
On scylla end, It's round robin per client ? by host ? I.e. multiple Clint from same host might need cycles here ?

Is it handled better on the java driver ?

Reply:
Oh, this aggressive :)

Yeah maybe 3 would be better lucky number here.

since you one error and you not necessarily do the cycle (and no guarantees for ordering anyhow)
On scylla end, It's round robin per client ? by host ? I.e. multiple Clint from same host might need cycles here ?

Is it handled better on the java driver ?

Reply:
Not sure it's better in Java.
Like I said, let's not change it this time.
If we see it becomes and issue we can fix this later.

Reply:
For the record, I did encounter this once, where running a few tests in a row, when trying to reproduce the tracing issue in 3.2.